app/controllers/buyers/applications_controller.rb

```diff
@@ -24,19 +25,16 @@ def index
     activate_menu :audience, :applications, :listing
 
     @states = Cinstance.allowed_states.collect(&:to_s).sort
-    accessible_services = current_account.accessible_services
-    @services = accessible_services.includes(:application_plans)
     @search = ThreeScale::Search.new(params[:search] || params)
-    @application_plans = current_account.application_plans.stock
-    @stock_and_custom_application_plans = current_account.application_plans.size
+    @services = accessible_services.includes(:application_plans)
 
     if params[:service_id]
-      @service = accessible_services.find params[:service_id]
+      @service = @services.find params[:service_id]
```

Comment:
After https://github.com/3scale/porta/pull/1798/files#r404886489, you may want to keep

```suggestion
      @service = accessible_services.find params[:service_id]
```

Reply:
I think this comment is still valid.